searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_iterator.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "searchiterator.h"
+#include "nearest_neighbor_distance_heap.h"
+#include <vespa/eval/tensor/dense/dense_tensor_view.h>
+#include <vespa/eval/tensor/dense/mutable_dense_tensor_view.h>
+#include <vespa/searchlib/fef/termfieldmatchdata.h>
+#include <vespa/searchlib/tensor/dense_tensor_attribute.h>
+#include <vespa/vespalib/util/priority_queue.h>
+#include <cmath>
+
+namespace search::queryeval {
+
+/**
+ * Search iterator for K nearest neighbor matching.
```

Comment:
Consider adding some more details about the implementation (e.g. brute force and how heap is used and updated).

Reply:
done.